src/cli/list.rs

```diff
@@ -0,0 +1,277 @@
+use std::path::PathBuf;
+
+use clap::Parser;
+use comfy_table::presets::NOTHING;
+use comfy_table::{Attribute, Cell, Color, ContentArrangement, Table};
+use human_bytes::human_bytes;
+use rattler_conda_types::{Channel, ChannelConfig, Platform};
+use rattler_lock::{LockedDependency, LockedDependencyKind};
+use serde::Serialize;
+
+use crate::lock_file::load_lock_file;
+use crate::project::SpecType;
+use crate::Project;
+
+// an enum to sort by size or name
+#[derive(clap::ValueEnum, Clone, Debug, Serialize)]
+pub enum SortBy {
+    Size,
+    Name,
+    Type,
+}
+
+/// List project's packages. Highlighted packages are explicit dependencies.
+#[derive(Debug, Parser)]
+#[clap(arg_required_else_help = false)]
+pub struct Args {
+    /// List only packages matching a regular expression
+    #[arg()]
+    pub regex: Option<String>,
+
+    /// The platform to list packages for. Defaults to the current platform.
+    #[arg(long)]
+    pub platform: Option<Platform>,
+
+    /// Whether to output in json format
+    #[arg(long)]
+    pub json: bool,
+
+    /// Whether to output in pretty json format
+    #[arg(long)]
+    pub json_pretty: bool,
+
+    /// Sorting strategy
+    #[arg(long, default_value = "name", value_enum)]
+    pub sort_by: SortBy,
+
+    /// The path to 'pixi.toml'
+    #[arg(long)]
+    pub manifest_path: Option<PathBuf>,
+}
+
+#[derive(Serialize)]
+struct PackageToOutput {
+    name: String,
+    version: String,
+    build: Option<String>,
+    size_bytes: Option<u64>,
+    kind: String,
+    source: Option<String>,
+    is_explicit: bool,
+}
+
+pub async fn execute(args: Args) -> miette::Result<()> {
+    // Load the project
+    let project = Project::load_or_else_discover(args.manifest_path.as_deref())
+        .ok()
+        .ok_or_else(|| miette::miette!("No project found"))?;
+
+    // Load the platform
+    let platform = args.platform.unwrap_or_else(Platform::current);
+
+    // Load the environment
+    // TODO: NOTE(hadim): make it an argument once environments are implemented.
+    // let environment = project.default_environment();
+
+    // Load the lockfile
+    let lockfile = load_lock_file(&project)
+        .await
+        .map_err(|_| miette::miette!("Cannot load lockfile. Did you run `pixi install` first?"))?;
+
+    let locked_deps = lockfile.packages_for_platform(platform).collect::<Vec<_>>();
+
+    // Get the explicit project dependencies
+    let project_dependency_names: Vec<String> = {
+        let dependencies = project
+            .default_environment()
+            .dependencies(Some(SpecType::Run), Some(platform));
```

Comment:
We currently combine all dependencies together, so you should specify `None` for the `SpecType` (this will combine them).